src/Storages/Statistics/Statistics.h

```diff
@@ -21,17 +19,21 @@ struct StatisticsUtils
     static std::optional<Float64> tryConvertToFloat64(const Field & value, const DataTypePtr & data_type);
 };
 
-/// Statistics describe properties of the values in the column,
+class ISingleStatistics;
+using SingleStatisticsPtr = std::shared_ptr<ISingleStatistics>;
+
+/// SingleTypeStatistics describe properties of the values in the column,
 /// e.g. how many unique values exist,
 /// what are the N most frequent values,
 /// how frequent is a value V, etc.
-class IStatistics
+class ISingleStatistics
```

Comment:
Hi @rschu1ze sorry, maybe I should describe my design a bit more clearly.

### Hierarchy of statistics

```
Statistics
    - ColumnStatistics
        - SingleStatistics
        - SingleStatistics
    - ColumnStatistics
       - SingleStatistics
       - SingleStatistics
```

`Statistics` can represent the statistics for
1. a part
2. multiple parts (but the the statistics e.g. `tdigest` for every part do not exist instead by a merged one)
3. a table
4. a query plan node

`ColumnStatistics` is for a single column(or expression) in `Statistics` and it can contains `minmax`, 'tdigest' etc. And now the column statistics can represent not only for a single part but also a merged one.

`SingleStatistics`: sushc as `minmax` `tdigest` etc. it represents a respect of a column.

### How merge work

`stat1` is for `part1` and `stat2` is for `part2`, if a query hits the 2 parts, then can merge together. Please notice how the marge works. For example `col_1` has `tdigest` statistics ,the two tdigest in `stat1` and `stat2` will be merged into a single one. This is quiet different with the old fashion.

Reply:
@rschu1ze `Statistics` is the most outer class I think it deserve the name what is your opinion?  I wonder whether we should rename `ISingleStatistics` back to `IStatistics` ?